flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumerThread.java

```diff
@@ -105,6 +117,8 @@ public void run() {
 
 					break;
 				} else {
+					Thread.sleep(idleMillisBetweenFetches);
```

Comment:
Agree. I'll change the default value of this to 0 in `KinesisConfigConstants`, and check value here before calling sleep.